package/MDAnalysis/analysis/nucleicacids.py

```diff
@@ -209,24 +253,147 @@ class WatsonCrickDist(NucPairDist):
+    c2_name: str (optional)
+        Name of Carbon 2 of nucleic acids
+        by default assigned to C2
+    g_name: str (optional)
+        Name of Guanine in topology
+        by default assigned to G
+    a_name: str (optional)
+        Name of Adenine in topology
+        by default assigned to G
+    u_name: str (optional)
+        Name of Uracil in topology
+        by default assigned to U
+    t_name: str (optional)
+        Name of Thymine in topology
+        by default assigned to T
+    c_name: str (optional)
+        Name of Cytosine in topology
+        by default assigned to C
+    **kwargs: dict
+        arguments for :class:`~MDAnalysis.analysis.base.AnalysisBase`
+
+    Attributes
+    ----------
+    results: numpy.ndarray
+        first index is selection second index is time
+
+    Raises
+    ------
+    ValueError
+        if the residues given are not amino acids
+    ValueError
+        if the selections given are not the same length
+
+    """
+
+    def __init__(self, strand1: List[ResidueClass], strand2: List[ResidueClass],
+                 o2_name: str = 'O2', c2_name: str = "C2",
+                 g_name: str = 'G', a_name: str = 'A', u_name: str = 'U',
+                 t_name: str = 'T', c_name: str = 'C',
+                 **kwargs) -> None:
+        selections: Tuple[List[mda.AtomGroup], List[mda.AtomGroup]] = self.select_strand_atoms(
+            strand1, strand2, c2_name, o2_name,
+            g_name=g_name, a_name=a_name,
+            t_name=t_name, u_name=u_name, c_name=c_name
+        )
+
+        super(MinorPairDist, self).__init__(selections[0], selections[1], **kwargs)
+
+
+
+class MajorPairDist(NucPairDist):
+    r"""Minor-Pair basepair distance for selected residues over a trajectory.
+
+    Takes two lists of :class:`~MDAnalysis.core.groups.Residue` objects and
+    calculates the Watson-Crick distance between them over the trajectory.
+    Bases are matched by their index in the lists given as arguments.
+
+    Parameters
+    ----------
+    strand1: List[Residue]
+        First list of bases
+    strand2: List[Residue]
+        Second list of bases
+    o6_name: str (optional)
+        Name of Oxygen 6 of nucleic acids
+        by default assigned to O6
+    n4_name: str (optional)
+        Name of Nitrogen 4 of nucleic acids
+        by default assigned to N4
+    g_name: str (optional)
+        Name of Guanine in topology
+        by default assigned to G
+    a_name: str (optional)
+        Name of Adenine in topology
+        by default assigned to G
+    u_name: str (optional)
+        Name of Uracil in topology
+        by default assigned to U
+    t_name: str (optional)
+        Name of Thymine in topology
+        by default assigned to T
+    c_name: str (optional)
+        Name of Cytosine in topology
+        by default assigned to C
+    **kwargs: dict
+        arguments for :class:`~MDAnalysis.analysis.base.AnalysisBase`
+
+    Attributes
+    ----------
```

Comment:
add docs for `times` (thanks to AnalysisBase)

Reply:
done